### PR TITLE
Fix precision error in float_parse_bytes

### PR DESCRIPTION
**Pull request by joesaah**
## Description
The purpose of this PR is to fix the issue discussed here: https://github.com/pydantic/pydantic/issues/11198.

## Changes
* Updated the `float_parse_bytes` using `lexical-parse-float` to handle all edge cases related to float parsing
* Added the failing scenario in the issue above as a test case
* I have also removed a doc comment from `lib.rs` to satisfy linting. Please let me know if you would like me to fix it some other way

## Testing
The test case I added with these changes can be added to the `main` branch. The test should fail. Compare that to running the test on this branch which should pass. Please let me know if you would like me to add more examples of previously failing cases to the tests here

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Implementation looks okay, but from my experience with jiter, we should just use [`lexical-parse-float`](https://crates.io/crates/lexical-parse-float) for float parsing - it's a very complex topic and `lexical-parse-float` covers all the edge cases.

**Comment by joesaah** [comment]
> @joesaah thanks so much for looking into this.
> 
> Implementation looks okay, but from my experience with jiter, we should just use [`lexical-parse-float`](https://crates.io/crates/lexical-parse-float) for float parsing - it's a very complex topic and `lexical-parse-float` covers all the edge cases.

@samuelcolvin thank you for your review and suggestion. I have add `lexical-parse-float` to the project and used it to parse floats in the problematic code. Apologies if anything looks funny in my code, this is my first time writing Rust 😅. I will be happy to make any other changes you or your team suggests

File: src/numbers.rs [hunk 1]
```diff
@@ -56,6 +58,16 @@ pub fn float_parse_str(s: &str) -> IntFloat {
 
 /// Parse bytes as an float.
 pub fn float_parse_bytes(s: &[u8]) -> IntFloat {
+    // If we know we're working with a float, let's go ahead and parse it with lexical-parse-float
+    // to handle all edge cases related to float parsing
+    if s.contains(&b'.') {
```
Comment (samuelcolvin) [inline src/numbers.rs:63]: this will require an extra scan over the bytes in all cases.

I think it would be better to keep the old logic of looking for a dot, then call `f64::from_lexical_with_options::<{ lexical_format::STANDARD }>` within the `if found_dot {` block.

It might slightly slower for floats, it's a smaller change and should be faster for ints which I think are more common.

**Review by samuelcolvin — COMMENTED** [review]
Sorry, one more change. Otherwise LGTM.

Amazing work if this is your first time writing rust!

File: src/numbers.rs (hunk 1 above)
Comment (joesaah) [reply src/numbers.rs]: Ah okay, I was unsure whether to do it this way or the way I went with. Ultimately, I just went with the way that had the better `make bench` output lol. Your reasoning totally makes sense, though, and I have made the suggested change. Thank you!

**Review by samuelcolvin — APPROVED** [review]
LGTM.

@davidhewitt please confirm you're happy with this.

File: src/numbers.rs (hunk 1 above)
Comment (samuelcolvin) [reply src/numbers.rs]: Thanks so much @joesaah, I generally practice make-bench-based-development, so I get where you're coming from.

**Review by davidhewitt — APPROVED** [review]
